runtime/format.cc

```diff
@@ -0,0 +1,330 @@
+//===-- runtime/format.cc ---------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "format.h"
+#include "../lib/common/format.h"
+#include "../lib/decimal/decimal.h"
+#include <limits>
+
+namespace Fortran::runtime {
+
+FormatControl::FormatControl(Context &context, const char *format,
```

Comment:
Do some of these functions need to be templatized to handle char(2) and char(4) cases?  If yes, I suppose that could come later.

Reply:
I should do it now to ensure that it can be done.

Reply:
Done!